test/unit/helpers/events.ts

```diff
@@ -265,6 +267,89 @@ export function createDummyBlockActionEventMiddlewareArgs(
   };
 }
 
+export function createDummyCustomFunctionMiddlewareArgs<
+  Options extends SlackEventMiddlewareArgsOptions = { autoAcknowledge: true },
+>(
+  data: {
+    callbackId?: string;
+    inputs?: Record<string, string | number | boolean>;
+    options?: Options;
+  } = { callbackId: 'reverse', inputs: { stringToReverse: 'hello' }, options: { autoAcknowledge: true } as Options },
+): SlackCustomFunctionMiddlewareArgs<Options> {
+  data.callbackId = data.callbackId ? data.callbackId : 'reverse';
+  data.inputs = data.inputs ? data.inputs : { stringToReverse: 'hello' };
+  data.options = data.options ? data.options : ({ autoAcknowledge: true } as Options);
+  const testFunction = {
+    id: 'Fn111',
+    callback_id: data.callbackId,
+    title: data.callbackId,
+    description: 'Takes a string and reverses it',
+    type: 'app',
+    input_parameters: [
+      {
+        type: 'string',
+        name: 'stringToReverse',
+        description: 'The string to reverse',
+        title: 'String To Reverse',
+        is_required: true,
+      },
+    ],
+    output_parameters: [
+      {
+        type: 'string',
+        name: 'reverseString',
+        description: 'The string in reverse',
+        title: 'Reverse String',
+        is_required: true,
+      },
+    ],
+    app_id: 'A111',
+    date_updated: 1659054991,
+    date_deleted: 0,
+    date_created: 1725987754,
+  };
+
+  const event = {
+    type: 'function_executed',
+    function: testFunction,
+    inputs: data.inputs,
+    function_execution_id: 'Fx111',
+    workflow_execution_id: 'Wf111',
+    event_ts: '1659055013.509853',
+    bot_access_token: 'xwfp-valid',
+  } as const;
+
+  const body = {
+    token: 'verification_token',
+    team_id: 'T111',
+    api_app_id: 'A111',
+    event,
+    event_id: 'Ev111',
+    event_time: 1659055013,
+    type: 'event_callback',
+  } as const;
+
+  if (data.options.autoAcknowledge) {
+    return {
+      body,
+      complete: () => Promise.resolve({ ok: true }),
```

Comment:
You can also set these fake methods as spies directly, so that you can then assert on whether/how they were called within tests. Same for `fail` and `ack`.

Reply:
I'm gonna push back on this because my implementation is not playing well with typescript when I try to do this 😞 

But their is deficiently a way to do this cleanly